addons/account/tests/test_account_payment.py

```diff
@@ -652,6 +652,27 @@ def test_payment_confirmation_with_bank_outstanding_account(self):
         payment.action_post()
         self.assertEqual(payment.state, 'paid')
 
+    def test_payment_state_with_unreconciliable_outstanding_account(self):
+        unreconciliable_account = self.env['account.account'].create({
+            'code': '209.01.01',
+            'name': 'Bank Account',
+            'account_type': 'asset_cash',
+            'reconcile': False,
+        })
+        self.company_data['default_journal_bank'].outbound_payment_method_line_ids[0].payment_account_id = unreconciliable_account
+        invoice = self.init_invoice(move_type='out_invoice', amounts=[10], post=True)
+
+        payment = self.env['account.payment.register']\
+            .with_context(active_model='account.move', active_ids=invoice.ids)\
+            .create({
+                'amount': 10.0, 
+                'journal_id': self.company_data['default_journal_bank'].id,
+                'payment_method_line_id': self.company_data['default_journal_bank'].outbound_payment_method_line_ids[0].id,
+                'currency_id': self.company_data['currency'].id})\
+            ._create_payments()
```

Comment:
```suggestion
        payment = self.env['account.payment.register'].with_context(
            active_model='account.move', 
            active_ids=invoice.ids,
        ).create({
            'amount': 10.0,
            'journal_id': self.company_data['default_journal_bank'].id,
            'payment_method_line_id': self.company_data['default_journal_bank'].outbound_payment_method_line_ids[0].id,
            'currency_id': self.company_data['currency'].id,
        })._create_payments()
```

Also, avoid `\` to continue the line, use parenthesis instead

https://peps.python.org/pep-0008/#maximum-line-length
> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.